docs/python/data-science-tutorial.md

```diff
@@ -0,0 +1,250 @@
+   ```
+
+   > **Note**: If you ever need to see the data type that has been used for a column, you can use the [DataFrame dtypes](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.dtypes.html#pandas.DataFrame.dtypes) attribute.
+
+1. Now that the data is in good shape, you can use [seaborn](http://seaborn.pydata.org/) and [matplotlib](https://matplotlib.org) to view how certain columns of the dataset relate to survivability. Add the following code to the next cell in your notebook and run it to see the generated plots.
+
+   ```python
+   import seaborn as sns
+   import matplotlib.pyplot as plt
+
+   fig, axs = plt.subplots(ncols=5, figsize=(30,5))
+   sns.violinplot(x="survived", y="age", hue="sex", data=data, ax=axs[0])
+   sns.pointplot(x="sibsp", y="survived", hue="sex", data=data, ax=axs[1])
+   sns.pointplot(x="parch", y="survived", hue="sex", data=data, ax=axs[2])
+   sns.pointplot(x="pclass", y="survived", hue="sex", data=data, ax=axs[3])
+   sns.violinplot(x="survived", y="fare", hue="sex", data=data, ax=axs[4])
+   ```
+
+   ![Graphing the titanic data](images/data-science-tutorial/jupyter-cell-02.png)
+
+   > **Note**: To better view details on the graphs, you can open them in plot viewer by hovering over the upper left corner of the graph and clicking the button that appears.
+
+1. These graphs are helpful in seeing some of the relationships between survival and the input variables of the data, but it’s also possible to use pandas to calculate correlations. To do so, all the variables used need to be numeric for the correlation calculation and currently gender is stored as a string. To convert those string values to integers, add and run the following code.
+
+   ```python
+   data.replace({'male': 1, 'female': 0}, inplace=True)
+   ```
+
+1. Now, you can analyze the correlation between all the input variables to identify the features that would be the best inputs to a machine learning model. The closer a value is to 1, the higher the correlation between the value and the result. Use the following code to correlate the relationship between all variables and survival.
+
+   ```python
+   data.corr().abs()[["survived"]]
+   ```
+
+   ![Determining the correlation between input variables and survival](images/data-science-tutorial/jupyter-cell-03.png)
+
+1. Looking at the correlation results, you’ll notice that some variables like gender have a fairly high correlation to survival, while others like relatives (sibsp = siblings or spouse, parch = parents or children) seem to have little correlation.
+
+   Let’s hypothesize that *sibsp* and *parch* are related in how they affect survivability, and group them into a new column called “relatives” to see whether the combination of them has a higher correlation to survivability. To do this, you will check if for a given passenger, the number of *sibsp* and *parch* is greater than 0 and, if so, yuo can then say that they had a relative on board.
```

Comment:
typo, should be "you" instead of "yuo" on the last line

Reply:
Good catch. Fixed.